.github/workflows/update-geth.yaml

```diff
@@ -0,0 +1,60 @@
+name: "Update celo-org/op-geth"
+on:
+  schedule:
+    - cron: "00 8 * * Mon"
+  workflow_dispatch:
+    inputs:
+      op-geth-base-branch:
+        description: "celo-org/op-geth base branch to update dependencies from"
+        required: false
+        default: "celo10"
+  pull_request:
+
+jobs:
+  job_id:
+    # Add "id-token" with the intended permissions.
+    permissions:
+      contents: "read"
+      id-token: "write"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Login at GCP Artifact Registry
+        uses: celo-org/reusable-workflows/.github/actions/auth-gcp-artifact-registry@v2.0
+        with:
+          workload-id-provider: "projects/1094498259535/locations/global/workloadIdentityPools/gh-optimism/providers/github-by-repos"
```

Comment:
@pputman-clabs could you give the github action GCP auth part a review?

I simply pasted it from another action and I don't know if this is the correct workload-id-provider.
The action requires authentication, although it simply reads some artifact properties from a public repository.
To open PR's, the action also requires some permissions in GitHub (see [here](https://github.com/peter-evans/create-pull-request?tab=readme-ov-file#workflow-permissions)). Would it be okay to enable this?